frontend/src/components/Projects/EditProject.js

```diff
@@ -1,101 +1,256 @@
-                     </div>
-                 </div>
-             </div>
-             <div class="container-fluid">
-                 <div className="form-group">
-                     <button
-                         className="btn btn-outline-primary btn-sm "
-                         onClick={handleSubmit}>
-                         Save
-                                 </button>
-                     <a href="/ProjPanel" className="btn btn-outline-primary btn-sm ml-4" role="button">Cancel</a>
-                 </div>
-             </div>
-         </div>
-     )
- }
-
- export default EditProject;
+import React, { Component } from 'react';
 
+class EditProject extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            projectUniqueNumber: "",
+            projectName: "",
+            duration: "",
+            budget: "",
+            peakInterval: "",
+            firstPV: "",
+            lastPV: "",
+            advPayment: "",
+            perfBond: "",
+            creditTime: "",
+            discountRate: ""
+        };
+    }
+    handleSubmit = () => {
+        const {
+            projectUniqueNumber,
+            projectName,
+            duration,
+            budget,
+            peakInterval,
+            firstPV,
+            lastPV,
+            advPayment,
+            perfBond,
+            creditTime,
+            discountRate
+        } = this.state;
+        // Invoke the passed in event callback
+
+        this.props.onSubmit({
+            projectUniqueNumber: projectUniqueNumber,
+            projectName: projectName,
+            duration: duration,
+            budget: budget,
+            firstPV: firstPV,
+            lastPV: lastPV,
+            perfBond: perfBond,
+            creditTime: creditTime,
+            discountRate: discountRate
+        });
+    }
+
+    componentDidMount() {
+        const {
+            projectUniqueNumber,
+            projectName,
+            duration,
+            budget,
+            peakInterval,
+            firstPV,
+            lastPV,
+            advPayment,
+            perfBond,
+            creditTime,
+            discountRate
+        } = this.props.location.state.project;
+
+        this.setState({
+            projectUniqueNumber,
+            projectName,
+            duration,
+            budget,
+            peakInterval,
+            firstPV,
+            lastPV,
+            advPayment,
+            perfBond,
+            creditTime,
+            discountRate
+        });
+
+    }
+
+    render() {
+        const { project } = this.props.location.state;
+
+        return (
+
+            <div className="card">
+                <div className="card-body">
+                    <h3 style={{ color: "black" }}>Project details</h3>
+                    <br></br>
+
+                        <div class="container-fluid">
+                            <div className="form-group col-md-5">
+
+
```

Comment:
Blank space, please remove. It shouldn’t be spaces between divs only one at the end of each code element. Just for have in mind :D